upup/pkg/fi/nodeup/command.go

```diff
@@ -637,3 +667,23 @@ func getNodeConfigFromServer(ctx context.Context, config *nodeup.ConfigServerOpt
 	}
 	return client.QueryBootstrap(ctx, &request)
 }
+
+func getAWSConfigurationMode(c *model.NodeupModelContext) (string, error) {
+
+	svc := c.Cloud.(awsup.AWSCloud).Autoscaling()
+
+	result, err := svc.DescribeAutoScalingInstances(&autoscaling.DescribeAutoScalingInstancesInput{
+		InstanceIds: []*string{&c.InstanceID},
+	})
+	if err != nil {
+		return "", fmt.Errorf("error describing instances: %v", err)
+	}
+	lifecycle := fi.StringValue(result.AutoScalingInstances[0].LifecycleState)
```

Comment:
I haven't checked, but I wonder if the lifecycle state is available via the instance metadata service? That would reduce the per-node API calls and avoid the IAM changes.

Reply:
It isn't, unfortunately. I passed on a request to AWS to add that.

Same goes for completing lifecycle hooks. See AWS own example here https://github.com/aws-samples/amazon-ec2-auto-scaling-group-examples/blob/main/features/lifecycle-hooks/userdata-managed-linux/template.yaml#L282